### PR TITLE
Support usage of lang attribute <html lang="fr"> as default language value

### PR DESCRIPTION
**Pull request by JaySunSyn**
- `<my-localize-elem use-html-language>` to use the lang attribute of the html tag. (would be cool if defaulted to true)
- `<my-localize-elem resource-url="/src/locals.json">` to auto load resource. (would be cool as well to be defaulted to this.resolveUrl('locales.json'))

Use `document.documentElement.lang = "de"` to change the language within the whole app.

What do you think?

### CODE REVIEW

File: app-localize-behavior.html [hunk 1]
```diff
@@ -164,12 +164,94 @@
     },
 
     /**
+     * If true, the global language attribute on the html tag
+     * will be used as the default language value.
+     * Use document.documentElement.lang = "de" to change the language 
+     * of the entire app.
+     */
+    useHtmlLanguage: {
```
Comment (notwaldorf) [inline app-localize-behavior.html]: I would call this `useLangAttribute`

File: app-localize-behavior.html [hunk 2]
```diff
@@ -164,12 +164,94 @@
...
+      type: Boolean,
+      value: false
+    },
+
+    /**
+     * If true, the browser language will be used per default.
+     * 'resources' have to include browser language key. 
+     * If not, HtmlLanguage will be used instead.
+     */
+    useBrowserLanguage: {
```
Comment (notwaldorf) [inline app-localize-behavior.html]: This one seems like too much magic, and I'd just use the other attribute.

File: bower.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "app-localize-behavior",
-  "version": "2.0.0",
+  "version": "2.1.1",
```
Comment (notwaldorf) [inline bower.json]: We do this in a separate change, when I cut a release

File: app-localize-behavior.html [hunk 4]
```diff
@@ -164,12 +164,94 @@
...
     bubbleEvent: {
       type: Boolean,
       value: false
+    },
+
+    /**
+     * The instance of the 'lang' attribute observer 
+     * of the html tag.
+     */
+    ___htmlLanguageObserver: {
```
Comment (notwaldorf) [inline app-localize-behavior.html]: Why does this observer need to exist as a property?

File: app-localize-behavior.html (hunk 1 above)
Comment (notwaldorf) [inline app-localize-behavior.html]: Instead, can you just check whether this element has a `lang` attribute, where it needs to be used? `if (this.language || this.getAttribute('lang') || html.getAttribute('lang')` or something. That way it's less user setup.

**Comment by JaySunSyn** [comment]
Thanks! Guess, we first need to agree on a concept.

We have 2 options here, the browser language `navigator.language` and the language set in the html tag which reflects the default app (default?) language.

Would you only pick one of those and if, which one?

I'd prefer to use the browser language and as a fallback the language set on the html tag. Would you agree?

**Comment by notwaldorf** [comment]
I think if I were to pick only one, it would be the language on the `html` tag. `navigator.language` isn't supported in IE11 and old Safari versions (so we can't use it)

**Pull request closed without merging**